odcockpit/src/plugins/environmentviewer/EnvironmentViewerGLWidget.cpp

```diff
@@ -391,6 +469,12 @@ namespace cockpit {
             }
 
             void EnvironmentViewerGLWidget::nextContainer(Container &c) {
+                
+                if(c.getDataType()==odcore::data::SharedPointCloud::ID()){
+                    velodyneFrame=c.getData<SharedPointCloud>();//Get shared point cloud
+                    velodyneSharedMemory=SharedMemoryFactory::attachToSharedMemory(velodyneFrame.getName());//Attach the shared point cloud to the shared memory           
```

Comment:
Can you check if you have already attached to the shared memory before to avoid re-attaching over and over again.